Add IPAM method for Infoblox (#2693)
theforeman/foreman-documentation: guides/common/modules/proc_installing-the-dhcp-infoblox-module.adoc

[id="Installing_the_DHCP_Infoblox_Module_{context}"]
= Installing the DHCP Infoblox Module

Use this procedure to install the DHCP Infoblox module on {SmartProxy}.

> Note that you cannot manage records in separate views.

Author: @ekohl Any ieda what this sentence means? If it's valiad, I would prefer to make it more specific: What records? And what views?

Reviewer: I don't know Infoblox that well, but I get the impression this should apply to the DNS module instead. There you can only manage a single view (chosen in the installer) per Smart Proxy. And since a domain is unique, you can't install multiple Smart Proxies (one for each view) and work around it that way.

Author: Let's keep it as-is for now.

You can also install DHCP and DNS Infoblox modules simultaneously by combining this procedure and xref:Installing_the_DNS_Infoblox_Module_{context}[].

.DHCP Infoblox Record Type Considerations
If you want to use both DHCP and DNS Infoblox modules, configure the DHCP Infoblox module only with the `fixedaddress` record type.
The `host` record type causes DNS conflicts and is not supported.

If you configure the DHCP Infoblox module with the `host` record type, you must unset both DNS {SmartProxy} and Reverse DNS {SmartProxy} options on your Infoblox-managed domains, because Infoblox does the DNS management by itself.

> You cannot use the `host` record type without creating conflicts and, for example, being unable to rename hosts in {Project}.

Reviewer 2: ```suggestion
Using the `host` record type leads to creating conflicts and being unable to rename hosts in {Project}.
```

--foreman-proxy-plugin-dhcp-infoblox-record-type fixedaddress \
--foreman-proxy-plugin-dhcp-infoblox-username admin
----

> . In the {ProjectWebUI}, navigate to *Infrastructure* > *{SmartProxies}*, select the {SmartProxy} with the Infoblox DHCP module, and click *Refresh*.
> . Ensure that the *dhcp* features are listed.

Reviewer: The installer should refresh, so I'd drop the refresh part.
```suggestion
. Optional: In the {ProjectWebUI}, navigate to *Infrastructure* > *{SmartProxies}*, select the {SmartProxy} with the Infoblox DHCP module, and ensure that the *dhcp* feature is listed.
```

> . For all domains managed through Infoblox, ensure that the DNS {SmartProxy} is set for that domain.

Reviewer: Plural?
```suggestion
. For all domains managed through Infoblox, ensure that the DNS {SmartProxy} is set for those domains.
```

Though I wonder why DNS is suddenly here. The installer command above sets up DHCP, not DNS. So there's no guarantee that it actually is there. Perhaps drop it and leave it for the Infoblox DNS module?

Author: I was wondering about the same, I'll move it to the DNS module.

> To verify, in the {ProjectWebUI}, navigate to *Infrastructure* > *Domains*, and inspect the settings of each domain.

Reviewer: This talks about Domains, but below it talks about Subnets.
```suggestion
To verify, in the {ProjectWebUI}, navigate to *Infrastructure* > *Subnets*, and inspect the settings of each subnet.
```

. For all subnets managed through Infoblox, ensure the following settings:
** The IP address management (IPAM) method of the subnet is set to `DHCP`.

> ** DHCP {SmartProxy} and Reverse DNS {SmartProxy} are set.

Reviewer: The DNS part is "special" here, like above.

I'd reduce this to:
```suggestion
** DHCP Proxy is set to your {SmartProxy}.
```

I don't know if this is branded different in Satellite, but at least in upstream it says "DHCP Proxy", not "DHCP Smart Proxy".

Author: Unfortunately, I have to stick with {SmartProxy}, because it is branded in Satellite.

+

> To verify, in the {ProjectWebUI}, navigate to *Infrastructure* > *Subnets*, and inspect the settings of each subnet.

Reviewer: This is redundant.

--foreman-proxy-plugin-dhcp-infoblox-record-type fixedaddress \

> --foreman-proxy-plugin-dhcp-infoblox-username admin

Reviewer: I think this (and password) need to be
```suggestion
--foreman-proxy-plugin-dhcp-infoblox-username _admin_
```

Also, you can drop `--foreman-proxy-dhcp-managed` since it's ignored when the provider is set to infoblox.

----
. Optional: In the {ProjectWebUI}, navigate to *Infrastructure* > *{SmartProxies}*, select the {SmartProxy} with the Infoblox DHCP module, and ensure that the *dhcp* feature is listed.
. In the {ProjectWebUI}, navigate to *Infrastructure* > *Subnets*.

> . For all subnets managed through Infoblox, ensure that the IP address management (IPAM) method of the subnet is set to `DHCP`.

Author: Should I file a feature request for implementing bulk changes for Subnets? And what about Domains, it's a similar case.

> . For all subnets managed through Infoblox, ensure that the IP address management (IPAM) method of the subnet is set to `DHCP`.

Reviewer: Yes, I think so.

> . For all subnets managed through Infoblox, ensure that the IP address management (IPAM) method of the subnet is set to `DHCP`.

Reviewer: @jeremylenz you may find this interesting. I wonder how much of the new all hosts page could be reused here.

> . For all subnets managed through Infoblox, ensure that the IP address management (IPAM) method of the subnet is set to `DHCP`.

Author: @ekohl @jeremylenz I wonder if it has a business value. How often does it happen that the user has more than one subnet or domain that need attributes configured the same way?
